custom/icds_reports/utils/aggregation_helpers/distributed/location_reassignment.py

```diff
@@ -0,0 +1,163 @@
+from dateutil.relativedelta import relativedelta
+from django.db import connections
+
+from corehq.apps.userreports.util import get_table_name
+from corehq.sql_db.connections import get_icds_ucr_citus_db_alias
+from custom.icds_reports.const import DASHBOARD_DOMAIN
+from custom.icds_reports.utils.aggregation_helpers import get_prev_table, transform_day_to_month
+
+
+class TempPrevTables(object):
```

Comment:
nit: maybe change to `TempPrevTablesBase` so it's clearer that this class is only meant to be used as a base class?

bonus: use `abc` to make it an actual abstract base classs